auth0/src/test/java/com/auth0/android/request/internal/OptionalCredentialsGsonTest.kt

```diff
@@ -0,0 +1,74 @@
+package com.auth0.android.request.internal
+
+import com.auth0.android.result.OptionalCredentials
+import com.google.gson.JsonParseException
+import org.hamcrest.MatcherAssert
+import org.hamcrest.Matchers
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Test
+import java.io.IOException
+import java.io.Reader
+import java.io.StringReader
+
+public class OptionalCredentialsGsonTest : GsonBaseTest() {
+    @Before
+    public fun setUp() {
+        gson = GsonProvider.gson
+    }
+
+    @Test
+    @Throws(Exception::class)
+    public fun shouldFailWithInvalidJson() {
+        Assert.assertThrows(JsonParseException::class.java) {
+            buildCredentialsFrom(json(INVALID))
+        }
+    }
+
+    @Test
+    public fun shouldNotFailWithEmptyJson() {
+        buildCredentialsFrom(json(EMPTY_OBJECT))
+    }
+
+    @Test
+    public fun shouldNotRequireAnyProperty() {
+        buildCredentialsFrom(StringReader("{\"scope\": \"openid profile\"}"))
+    }
+
+    @Test
+    @Throws(Exception::class)
+    public fun shouldReturnLegacy() {
+        val credentials = buildCredentialsFrom(json(LEGACY_CREDENTIALS))
```

Comment:
tests that the legacy json file is still parsed correctly (with multiple missing / nulls values)